models/Tweet.js

```diff
@@ -15,36 +27,35 @@ export default class Tweet {
 
   likers(tweet, { lastCreatedAt = 0, limit = 10 }) {
     return this.context.User.collection.find({
-      likedIds: tweet.id,
+      likedIds: tweet._id,
       createdAt: { $gt: lastCreatedAt },
     }).sort({ createdAt: 1 }).limit(limit).toArray();
   }
 
   async insert(doc) {
-    // XXX: proper id generation strategy
-    const id = (await this.collection.find().count()).toString();
     const docToInsert = Object.assign({}, doc, {
-      id,
       createdAt: Date.now(),
       updatedAt: Date.now(),
     });
-    await this.collection.insert(docToInsert);
-    this.pubsub.publish('tweetInserted', docToInsert);
+    const id = (await this.collection.insertOne(docToInsert)).insertedId;
+    this.pubsub.publish('tweetInserted', await this.findOneById(id));
     return id;
   }
 
   async updateById(id, doc) {
-    const ret = await this.collection.update({ id }, {
+    const ret = await this.collection.update({ _id: id }, {
       $set: Object.assign({}, doc, {
         updatedAt: Date.now(),
       }),
     });
+    this.loader.clear(id);
```

Comment:
The loader caches by id, in our case for the entirety of the request. So there's probably ways you could break it, it seems safer to have this.